google-cloud-bigquery/src/main/scala/akka/stream/alpakka/googlecloud/bigquery/model/ErrorProtoJsonProtocol.scala

```diff
@@ -22,25 +22,25 @@ object ErrorProtoJsonProtocol {
    * @param location specifies where the error occurred, if present
    * @param message A human-readable description of the error
    */
-  final case class ErrorProto(reason: String, location: Option[String], message: String) {
+  final case class ErrorProto(reason: Option[String], location: Option[String], message: Option[String]) {
 
     @silent("never used")
     @JsonCreator
-    private def this(@JsonProperty(value = "reason", required = true) reason: String,
+    private def this(@JsonProperty(value = "reason") reason: String,
                      @JsonProperty("location") location: String,
-                     @JsonProperty(value = "message", required = true) message: String) =
-      this(reason, Option(location), message)
+                     @JsonProperty(value = "message") message: String) =
+      this(Option(reason), Option(location), Option(message))
 
-    def getReason = reason
+    def getReason = reason.asJava
     def getLocation = location.asJava
-    def getMessage = message
+    def getMessage = message.asJava
 
-    def withReason(reason: String) =
-      copy(reason = reason)
+    def withReason(reason: util.Optional[String]) =
+      copy(reason = reason.asScala)
     def withLocation(location: util.Optional[String]) =
       copy(location = location.asScala)
-    def withMessage(message: String) =
-      copy(message = message)
+    def withMessage(message: util.Optional[String]) =
+      copy(message = message.asScala)
```

Comment:
Sorry, I'm confused, which accessors are missing?

Reply:
The `with` methods all accept `java.util.Optional`. We should have overloads that take `Option` as well.

Reply:
I see, I thought that `with` methods are Java-only APIs since Scala case classes have `copy` available, is this not the case? If so, I need to fix this in many many places ...

Reply:
In Akka core as well as other Akka projects, types that have Scala and Java APIs are not usually defined as case classes, because the extra methods they generate will break bin compatibility whenever fields are changed (though perhaps not code compatibility for Scala users who recompile against new deps). It's safer to have a class where the accessors are controlled by us so that we can make additive changes easily and control deprecations more explicitly. We don't allow bin compat breaking changes in patch versions, but we also try to make a best effort across all upgrades to avoid it as well.

A common scenario is when users depend on multiple Alpakka connectors in the same project. We want to have some overlap of compatibility across versions to give users a chance to upgrade and deprecating old APIs. Since there have been no releases that have deprecated the google config across the connectors this common library will replace we have to be careful not to break the upgrade experience since there was no deprecation forewarning.

Reply:
Thank you for the detailed explanation! This was very helpful and would be a great addition to the contributing guide.

Reply:
BTW I added Scala overrides for `with` methods for the `GoogleSettings` class and family.